exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/Foreman.java

```diff
@@ -419,6 +425,64 @@ private void runPhysicalPlan(final PhysicalPlan plan) throws ExecutionSetupExcep
     logger.debug("Fragments running.");
   }
 
+  /**
+   * This is a helper method to run query based on the list of PlanFragment that were planned
+   * at some point of time
+   * @param fragmentsList
+   * @throws ExecutionSetupException
+   */
+  private void runFragment(List<PlanFragment> fragmentsList) throws ExecutionSetupException {
+    // need to set QueryId, MinorFragment for incoming Fragments
+    PlanFragment rootFragment = null;
+    boolean isFirst = true;
+    final List<PlanFragment> planFragments = Lists.newArrayList();
+    final int fragmentsCount = fragmentsList.size();
+    for (PlanFragment myFragment : fragmentsList) {
+      final FragmentHandle handle = myFragment.getHandle();
+      // for split plan number of minor fragments will be always one,
+      // but minor fragmentId may not be 0, as it is one of the minor fragments from split plan
+      final int minorFragment = (fragmentsCount == 1) ? 0 : handle.getMinorFragmentId();
```

Comment:
Even though I understand that a split plan consists of a single minor fragment, I wonder if there is a better way to handle minor fragment numbering. Why do not we just re-number minor fragments during planning/splitting phase at the first place instead of relying on fragmentsCount == 1 check?